Fix netsocket DNS test for IAR

jarvte opened the pull request:
### Description

Test case printed IP address. If ip address is null, IAR compiled binary fails.
Added check for printing null. If IP address is null, then it prints string 'null'.

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [X] Test update
    [ ] Breaking change

### Reviewers

@kimlep01 @kivaisan

Commit:
Fix netsocket DNS test for IAR

Test case printed IP address. If ip address is null, IAR compiled binary fails.
Added check for printing null. If IP address is null, then it prints string 'null'.

kivaisan commented on TESTS/netsocket/dns/main.cpp:
```diff
@@ -160,7 +160,7 @@ static void net_bringup()
     nsapi_error_t err = net->connect();
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
-    printf("MBED: IP address is '%s'\n", net->get_ip_address());
+    printf("MBED: IP address is '%s'\n", net->get_ip_address() ? net->get_ip_address() : "null");
```
There should be an IP address after successful connect. Could you check why it is not returned?

jarvte replied:
After connect there might not be an ip address available from the modem. I Added retry logic to get_ip_address() in second commit. Now it works. Good catch!

Commit:
Cellular: Added retry logic for getting ip address from stack

If AT_CellularStack::get_ip_address() is called via NetworkInterface
right after connect there might not be an ip address. After testing
in iot network there is no retry logic and ip address is given even
right after connect is finished.

kivaisan commented on features/cellular/framework/AT/AT_CellularStack.cpp:
```diff
@@ -59,30 +60,38 @@ const char *AT_CellularStack::get_ip_address()
+        if (_at.info_resp()) {
+            _at.skip_param();
+
+            int len = _at.read_string(_ip, NSAPI_IPv4_SIZE);
```
I think there is a bug in original code which could be fixed with this PR:

> In dual-stack terminals (<PDP_type> IPV4V6), the IPv6 address will be provided in <PDP_addr_2>. For terminals with a single IPv6 stack (<PDP_type> IPV6) or due to backwards compatibility, the IPv6 address can be provided in parameter <PDP_addr_1>.

jarvte replied:
force-pushed fix
